### PR TITLE
[CI] Only run git diff once

### PR DESCRIPTION
**Pull request by alichtman**
We can return the exit code from the same command we print the diff
from.

### CODE REVIEW

File: .github/workflows/documentation.yaml [hunk 1]
```diff
@@ -46,18 +46,14 @@ jobs:
...
-          git --no-pager diff
-          if git --no-pager diff --quiet
+          if git --no-pager diff --exit-code
           then
             echo "✅ The zshelldocs are up to date"
           else
             {
               echo "❌ The zshelldocs in the repo are not up to date"
               echo 'Please regenerate them with $ make doc'
-            } >&2
```
Comment (pschmitt) [inline .github/workflows/documentation.yaml:61]: This wasn't intentional I assume.

Comment (alichtman) [reply .github/workflows/documentation.yaml:61]: It was. It spits out the git diff output in all cases, allowing the omission of the second git diff command just to get the return code.

Comment (pschmitt) [reply .github/workflows/documentation.yaml:61]: I meant this line. Why not show the error message on stderr?

**Comment by pschmitt** [comment]
The test is failing because there *is* a diff. The issue here is with `make docs` ie. `zsd`, not git diff.

**Comment by alichtman** [comment]
I know. This is a CI optimization PR, not a fix for the broken tests

**Pull request merged**